### DISCUSSION TITLE
Assembly not resolve

### DISCUSSION THREAD

**Discussion by tonisimakov99** [icsharpcode/ILSpy#3107] [Help]
### Steps to reproduce
```
var resolver = new UniversalAssemblyResolver("System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e", true, null, null);
                resolver.Resolve(AssemblyNameReference.Parse("System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e"));
```

### Error message shown

```
ICSharpCode.Decompiler.Metadata.ResolutionException
  HResult=0x80131500
  Message=Failed to resolve assembly: 'System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e'
Resolve result: <not found>
  Source=ICSharpCode.Decompiler
  StackTrace:
   at ICSharpCode.Decompiler.Metadata.UniversalAssemblyResolver.ResolveInternal(IAssemblyReference name)
   at ICSharpCode.Decompiler.Metadata.UniversalAssemblyResolver.FindAssemblyFile(IAssemblyReference name)
   at ICSharpCode.Decompiler.Metadata.UniversalAssemblyResolver.Resolve(IAssemblyReference name)
   at BoxEngineTests.ShaderTests.Some() in D:\BoxEngine\BoxEngineTests\ShaderTests.cs:line 330
```

### Details
* Product in use: ICSharpCode.Decompiler  nuget
* Version in use: 8.1.0.7455

**Comment by siegfriedpammer** [comment]
The first parameter of the `UniversalAssemblyResolver` constructor is named `mainAssemblyFileName`, it needs to point to an actual file on the file system or should be omitted. The second parameter reads `throwOnError` which is the exact behavior you are seeing. The assembly cannot be resolved due to a misconfiguration of `UniversalAssemblyResolver` and throws an error (as configured).

What are you trying to do?

**Comment by tonisimakov99** [comment]
> The first parameter of the `UniversalAssemblyResolver` constructor is named `mainAssemblyFileName`, it needs to point to an actual file on the file system or should be omitted. The second parameter reads `throwOnError` which is the exact behavior you are seeing. The assembly cannot be resolved due to a misconfiguration of `UniversalAssemblyResolver` and throws an error (as configured).
> 
> What are you trying to do?

I tried setting `mainAssemblyFileName = null` and `throwOnError = false`, but I get `PEFile = null` after `Resolve()`. I'm trying to traverse a method call tree and convert their bodies to SPIR-V.
